scipy/stats/_stats_py.py

```diff
@@ -8139,44 +8150,67 @@ def combine_pvalues(pvalues, method='fisher', weights=None):
 
     Notes
     -----
-    Fisher's method (also known as Fisher's combined probability test) [1]_ uses
-    a chi-squared statistic to compute a combined p-value. The closely related
-    Stouffer's Z-score method [2]_ uses Z-scores rather than p-values. The
-    advantage of Stouffer's method is that it is straightforward to introduce
-    weights, which can make Stouffer's method more powerful than Fisher's
-    method when the p-values are from studies of different size [6]_ [7]_.
-    The Pearson's method uses :math:`log(1-p_i)` inside the sum whereas Fisher's
-    method uses :math:`log(p_i)` [4]_. For Fisher's and Pearson's method, the
-    sum of the logarithms is multiplied by -2 in the implementation. This
-    quantity has a chi-square distribution that determines the p-value. The
-    `mudholkar_george` method is the difference of the Fisher's and Pearson's
-    test statistics, each of which include the -2 factor [4]_. However, the
-    `mudholkar_george` method does not include these -2 factors. The test
-    statistic of `mudholkar_george` is the sum of logisitic random variables and
-    equation 3.6 in [3]_ is used to approximate the p-value based on Student's
-    t-distribution.
+    All methods assume tests based on continuous distributions, i.e., the
+    distribution of p-values under the null hypothesis must be the continuous
+    uniform distribution on the interval [0,1]. If this function is applied to
+    tests with a discrete statistics such as any rank test or contingency-table
+    test, it will yield systematically wrong results, e.g. Fisher's method will
+    systematically overestimate the p-value [1]_. This problem becomes less
+    severe for large sample sizes when the discrete distributions become
+    approximately continuous.
+
+    The differences between the methods can be best illustrated by their
+    statistics and what aspects of a combination of p-values they emphasise
+    when considering significance [2]_. For example, methods emphasising large
+    p-values are more sensitive to strong false and true negatives; conversely
+    methods focussing on small p-values are sensitive to positives.
+
+    * Fisher's method (also known as Fisher's combined probability test) [3]_
+      uses the product of individual p-values: :math:`\\prod_i p_i`.
+      (Mind that this product is not the combined p-value.)
+      This method emphasises small p-values.
+      Note that the test statistic used internally and returned by this method
+      is :math:`-2\\sum_i \\log(p_i)` for numerical reasons.
```

Comment:
I think it's actually of interest that this quantity follows a 𝝌² distribution. It's the historical reason why Fisher's exact test was adopted, despite having some defects such as being sensitive to very small p-values (as you point out). I think you should still mention it somewhere, otherwise this formula seems unmotivated.

Reply:
I added this aspect back, however not that prominently and only after introducing the test

Also, there is a straightforward motivation for Fisher’s method beyond historical practical considerations: If you want to know how unlikely two events are to coincide, you multiply their probabilities.

Reply:
> Also, there is a straightforward motivation for Fisher’s method beyond historical practical considerations: If you want to know how unlikely two events are to coincide, you multiply their probabilities.

I'm not so sure. This is what Heard and Rubin-Delanchy had to say on the matter

*However, little attention has been afforded to highlighting precisely the settings in which these
combiners are optimal, and thereby determining their suitability for an application at hand.
Instead, the widespread adoption of the above statistics for combining p-values can be
largely attributed to their simplicity and mathematical convenience: Under H0, −2SP and
−2SF are both distributed as χ2 2n; SG and SE have slightly more awkward closed-form
distributions which can be well approximated by Gaussian distributions for large n; SS
is N(0, n); and ST ∼ Beta(1, n). A meta-analysis combined p-value is therefore trivial
to obtain for each of these statistics, but they can differ substantially.*

Also I think explaining Fisher's method by saying if you want to know how unlikely two events are to coincide, you
multiply their probabilities is one of those intuitive but incomplete explanations common in the sciences. With
meta-analysis methods like these, what one typically wants is to be able to aggregate p-values from multiple studies, which may themselves be weakly powered to get a result that is more strongly powered. It does not appear obvious to me that the probability that the null hypothesis was incorrectly rejected for every test is the best statistic for performing meta-analysis of multiple studies. I really think the reason Fisher's method became among competing methods is due to its ease of computation to the the nice way the distribution works out.

Reply:
No disagreement here, hence *motivation* and not *watertight rationale.* There are clear downsides to Fisher’s method, such as completely ignoring if some studies point significantly in the other direction.